subcommands/targets/offline-update.go

```diff
@@ -179,11 +182,24 @@ func downloadTufRepo(factory string, tag string, prod bool, dstDir string) error
 		ver += 1
 	}
 
-	metadataFileNames := []string{
-		"timestamp.json", "snapshot.json", "targets.json",
+	meta, err := api.TufTargetMetadataRefresh(factory, target, tag, expiresIn, prod)
+	if err != nil {
+		return err
+	}
+	metadataNames := []string{
+		"timestamp", "snapshot", "targets",
 	}
-	for _, file := range metadataFileNames {
-		err = downloadMetadataFile(file)
+	for _, metaName := range metadataNames {
+		f, err := os.Create(path.Join(dstDir, metaName+".json"))
```

Comment:
I think [os.WriteFile](https://pkg.go.dev/os#WriteFile) would save you some a few lines of code and error handling.

Reply:
done, forced push.